### DISCUSSION TITLE
Weird clash between a PixiJS element and other element using `translateZ(0)` in Safari 16.2

### DISCUSSION THREAD

**Discussion by rickcurran** [pixijs/pixijs#9178] [Q&A] [answered]
Hi,
 I'm trying to implement a shape animation background using PixiJS, basically along the lines of this [example tutorial](https://georgefrancis.dev/writing/create-a-generative-landing-page-and-webgl-powered-background/). 

I've got the example working on my own site but I also have some elements on the page that have a hover-effect which uses `translateZ(0)` in the CSS and some of these elements are showing weird distortion / effects when hovering, so it seems as there is a clash between them somehow. This issue is only happening in Safari, Chrome and Firefox are unaffected so it seems like this might be a Safari bug.

One other possible contributing factor is that the example I'm working with is using PixiJS version 5, so it's possible the issue doesn't occur in version 7. However I tried switching to v7 but then got the following error:

`Error: [Package Error] "@pixi/core" no dependency version info found. (Imported by "@pixi/mixin-cache-as-bitmap").` 

I did look up that error to see if I could resolve it but I couldn't really see how to resolve it. So I thought the first thing I should see is if there is any kind of known issue for this sort of glitch. You can see a preview of the weirdness on this url:

https://neon.qreate.co.uk/static/test.html

If you scroll down you'll see a panel with an image with green background / red figure, hover over that and you should see the glitch effect.

I am using Safari 16.2 on MacOS 12.6 Monterey.

(Also, if it's not obvious I'm completely new to PixiJS so have very little experience with it!)

**Comment by SuperSodaSea** [comment]
For the `[Package Error]` problem: PixiJS v7 does not work with Skypack currently, see #8822 and #8798 for details.

**Reply by rickcurran** [reply]
@SuperSodaSea Thanks for replying! Ah ok. I am importing PixiJS like this in my JS file:

`import * as PIXI from "https://cdn.skypack.dev/pixi.js@5.x";`

Is there an alternative CDN version that I can use? Or will I need to look into compiling it all locally?

**Reply by SuperSodaSea** [reply]
Try `https://cdn.jsdelivr.net/npm/pixi.js@7.x/dist/pixi.min.js` or `https://unpkg.com/pixi.js@7.x/dist/pixi.min.js`, these bundles are provided in a single file and won't have dependency problems. Skypack provides packages in separate ESM files, and it needs `peerDependencies` information in package.json to work properly; unfortunately in PixiJS v7 we removed `peerDependencies` (see #8630), so Skypack can't find the dependencies correctly. You can also use tools like Webpack, Rollup or Vite to compile it locally, which may require a little more configuration work.

(For the Safari problem, I don't have an Apple device myself, so sorry for not being able to help you.)

**Comment by rickcurran** [comment]
@SuperSodaSea Thank you, that is very helpful. I'll try the latest version and see if I still encounter the glitch I'm seeing.

**Comment by rickcurran** [comment, marked as answer]
Ok, I have been able to get the latest V7 PixiJS code running now, I was still seeing the glitch in Safari when hovering over the elements even after getting the latest version working. 

However, after a bit of trial and error removing some of the CSS `transform` styles on the hoverable element that was showing the glitch and it helped me narrow down the issue to be related to the `rotateX` and `rotateY` transforms, which then with a bit of googling ("css transform causing glitch with webgls canvas in safari") I found [a Stackoverflow post](https://stackoverflow.com/questions/74498305/css-transform-spin-animation-glitching-in-safari-ios) which had the fix I needed. 

Basically I added `transform: translateZ(100px);  transform-style: preserve-3d;` to the parent element that contains the hoverable element, I think this basically brings the element forwards / upwards in the Z axis which stops the hoverable element and the Pixi canvas elements from interacting.

Thanks again @SuperSodaSea for the help on getting the V7 PixiJS working.
